client/src/components/profileInfoEdit/ProfileInfoEdit.js

```diff
@@ -64,14 +68,91 @@ const ProfileInfoEdit = () => {
     }
   };
 
+  const setPassword = (field, value) => {
+    setprofileDataPassword({
+      ...profileDataPassword,
+      [field]: value,
+    });
+    if (errors[field]) {
+      setErrors({
+        ...errors,
+        [field]: null,
+      });
+    }
+  };
+
+  const passwordSchema = yup.object().shape({
+    password: yup
+      .string()
+      .required('Please Enter your password')
+      .matches(
+        /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/,
+        'Use 8 or more characters with a mix of letters, numbers & symbols',
+      ),
+    confirmedPassword: yup
+      .string(),
+  });
+
+  const checkPassword = async () => {
+    const { password, confirmedPassword } = profileDataPassword;
+
+    const valid = await passwordSchema.isValid({
+      password,
+      confirmedPassword,
+    });
+    if (valid) {
+      return true;
+    }
+    return false;
+  };
+
+  const findFormErrors = async () => {
```

Comment:
Since you are using fomilk and you have already put the error message inside the schema (line 84), I think it will automatically handle that once the form is submitted, so you won't need this extra function.

Reply:
hi, i am and will not be using formik, the schema is only to do password validation, yup and formik are different things and i havent linked yup to formik in this page

Reply:
Wouldn't it is not maintainable as you are currently handling the error message using both yup and your own custom `findFormErrors` function? For example, if you need another validation rule for the password, you'll need to add it in the schema first and also add the checking logic in the `findFormErrors`

Reply:
Hi, constraints to the password are done in the yup schema to avoid checking separately and everything else done in function, so to add another rule to pwd, just changing schema is fine, or just change function if related to other variables

Reply:
As the schema checks all the constraints and then tells you if it's valid